pyvex/lifting/util/lifter_helper.py

```diff
@@ -136,11 +138,13 @@ def pp_disas(self):
         insts = self.disassemble()
         for addr, name, args in insts:
             args_str = ",".join(str(a) for a in args)
-            disasstr += f"{addr:0#8x}:\t{name} {args_str}\n"
+            disasstr += f"{addr:#08x}:\t{name} {args_str}\n"
         print(disasstr)
 
     def error(self):
         return self.errors
 
     def disassemble(self):
-        return self.lift(disassemble=True)
+        if self.disassembly is None:
+            self.lift(self.data, disasm=True)
+        return self.disassembly
```

Comment:
Introduced a new variable `self.disassembly` here to save the disasm result, since `Lifter.lift()` returns irsb, when disassembly is returned from `GymratLifter`'s overriden `_lifter`, it won't be passed back to the caller.